client/src/objects/terrain.js

```diff
@@ -3,7 +3,9 @@ import Phaser from 'phaser';
 export class Terrain extends Phaser.GameObjects.Sprite {
```

Comment:
This one would probably be better if inherited from a group and add objects in the group.
`this.scene.physics.add.collider` also supports group as a parameter.